webapp/components/test-runs-query-builder.html

```diff
@@ -7,38 +7,297 @@
 <link rel="import" href="../bower_components/polymer/polymer-element.html">
 <link rel="import" href="../bower_components/polymer/lib/elements/dom-if.html">
 <link rel="import" href="../bower_components/polymer/lib/elements/dom-repeat.html">
+<link rel="import" href="../bower_components/paper-icon-button/paper-icon-button.html">
+<link rel="import" href="../bower_components/iron-icons/iron-icons.html">
+<link rel="import" href="../bower_components/iron-icons/editor-icons.html">
+<link rel="import" href="../bower_components/paper-button/paper-button.html">
+<link rel="import" href="../bower_components/paper-card/paper-card.html">
+<link rel="import" href="../bower_components/paper-dropdown-menu/paper-dropdown-menu.html">
+<link rel="import" href="../bower_components/paper-input/paper-input.html">
+<link rel="import" href="../bower_components/paper-item/paper-item.html">
+<link rel="import" href="../bower_components/paper-listbox/paper-listbox.html">
 <link rel="import" href="./display-logo.html">
+<link rel="import" href="./product-info.html">
 <link rel="import" href="./test-runs-query.html">
 
 <dom-module id="test-runs-query-builder">
   <template>
-    <h3>Products</h3>
+    <style>
+      #add-button {
+        background-color: var(--paper-blue-500);
+        color: white;
+      }
+      #clear-button {
+        background-color: var(--paper-red-500);
+        color: white;
+      }
+    </style>
+
+    <h3>
+      Products
+      <paper-icon-button icon="editor:mode-edit"
+                         onclick="[[toggleEdit]]"></paper-icon-button>
+    </h3>
+    <template is="dom-if" if="[[debug]]">
+      <template is="dom-repeat" items="[[productSpecs]]" as="p" index-as="i">
+        [[p]]
+      </template>
+    </template>
     <div>
-      <template is="dom-repeat" items="[[products]]" as="product">
-        <display-logo product="[[product]]"></display-logo>
-        [[product.browser_name]]
-        <template is="dom-if" if="[[product.browser_version]]">
-          [[product.browser_version]]
-        </template>
+      <template is="dom-repeat" items="[[products]]" as="p" index-as="i">
```

Comment:
Note that nested fields on an object, which is inside an array, aren't observable from the parent, which is why they fields are separated in `product-builder`.